app/controllers/concerns/hal_actions/actions.rb

```diff
@@ -0,0 +1,104 @@
+module HalActions::Actions
+  def index
+    respond_with index_collection, index_options
+  end
+
+  def show
+    respond_with show_resource, show_options
+  end
+
+  def create
+    create_resource.tap do |resource|
```

Comment:
I would probably not use `resource` as the block arg name, just b/c of the confusion that we also have a `resource` method and so that could be slightly confusing to a less familiar reader, but obviously it is fine and safe thing to do.